activerecord/lib/active_record/secure_token.rb

```diff
@@ -40,9 +40,10 @@ module ClassMethods
       #   The callback when the value is generated. When called with <tt>on:
       #   :initialize</tt>, the value is generated in an
       #   <tt>after_initialize</tt> callback, otherwise the value will be used
-      #   in a <tt>before_</tt> callback. It will default to <tt>:create</tt>.
-      #
-      def has_secure_token(attribute = :token, length: MINIMUM_TOKEN_LENGTH, on: :create)
+      #   in a <tt>before_</tt> callback. It will default to the value of
+      #   <tt>config.active_record.generate_secure_token_on</tt>. As of Rails
+      #   7.1, that configuration option defaults to <tt>:initialize</tt>.
```

Comment:
```suggestion
      #   in a <tt>before_</tt> callback. When not specified, +:on+ will use the value of
      #   <tt>config.active_record.generate_secure_token_on</tt>, which defaults to +:initialize+
      #   starting in \Rails 7.1
```

I moved some words around to try to emphasize that the default changed in 7.1, which I think prevents having to specify the old value because it feels more implied?